### ISSUE TITLE
doubt about validation dataset

### ISSUE THREAD

**Issue by bedman367** [Anirudh257/strm#5]
Hi! Thanks for your excellent work!  I'm  intresting in few-shot action recognition too. But it confused me a lot that how should I find the best iteration during meta-training phase with validation dataset ? Your work is based on TRX ,which I 've tried to reproduce ,and I find that both TRX and STRM don't use validation dataset explicitly in the code, instead give the final iteration directly.
In my own experiment, I found that the test acc on the val dataset is lower than the test acc on the test dataset,  and if I set the frequency of the validation to 1000 meta-traing tasks  I got val acc that fluctuates up and down.  Have you met this question during your  experiment?  
the works in this area are so few that I can't find a reasonable explanation, so I would be very appreciated if you can solve my confusion...

**Comment by Anirudh257** [comment]
Hi @bedman367, I understand your doubt. Please follow: https://github.com/tobyperrett/trx/issues/3 and let me know if you need any help. My code is based on this.

**Comment by bedman367** [comment]
> Hi @bedman367, I understand your doubt. Please follow: [tobyperrett/trx#3](https://github.com/tobyperrett/trx/issues/3) and let me know if you need any help. My code is based on this.


Thanks so much for your quickly reply!  I 've noticed this issue , so did you determine the best iteration in the way you mentioned? https://github.com/tobyperrett/trx/issues/3#issuecomment-811824358
Actually , I only want to find the best iteration with validation dataset and don't want to retrain on the train/val before testing..
By adapting the code , I indeed implement what I 've expected. So my questions can ultimately be boiled down to two points:
1. Have you used validation dataset to find the best iterations?
2. The evaluation accuracy on the test set is higher than the evaluation accuracy on the validation set ，is this normal ?

**Comment by Anirudh257** [comment]
1) Yes, I found that to be 75K iterations for the SSv2 dataset and other datasets have different numbers.

2) It is normal.

**Comment by bedman367** [comment]
Thanks for your answer, I have no more questions. Hope everything goes well with your work！

**Comment by Anirudh257** [comment]
No issues. Please reopen the issue whenever you have doubts.
